Dockerfile

```diff
@@ -29,20 +29,27 @@ FROM golang:1.11.2 AS base
 ARG APT_MIRROR=deb.debian.org
 RUN sed -ri "s/(httpredir|deb).debian.org/$APT_MIRROR/g" /etc/apt/sources.list
 
-FROM base AS criu
-# Install CRIU for checkpoint/restore support
-ENV CRIU_VERSION 3.6
-# Install dependency packages specific to criu
+
+# Just a little hack so we don't have to install these deps twice, once for runc and once for dockerd
+FROM base AS runtime-dev
 RUN apt-get update && apt-get install -y \
+	libapparmor-dev \
+	libseccomp-dev \
+# Install dependency packages specific to criu
 	libnet-dev \
 	libprotobuf-c0-dev \
 	libprotobuf-dev \
 	libnl-3-dev \
 	libcap-dev \
 	protobuf-compiler \
```

Comment:
Are the protobuf things needed for all off `runc`, `criu` _and_ `docker` ?

(Also, are they needed at _runtime_ or only ad build-time)?

Reply:
Thanks for the feedback, not all of those packages are needed at _runtime_.
Actually, `criu` is linked only with `librt`, `libprotobuf-c`, `libdl`, `libnl-3` and `libnet`.
Without these installed at _runtime_ it fails, for example, with:
```
$ criu --version
criu: error while loading shared libraries: libprotobuf-c.so.1: cannot open shared object file: No such file or directory
```
I will update the commit to install only the necessary packages.

Reply:
Actually, instead of installing these packages at runtime, it would be better to only copy the required files (i.e. `libnet.so.1`, `libnl-3.so.200`, `libprotobuf-c.so.1`). I will update the commit accordingly.

Reply:
> it would be better to only copy the required files

@rst0git it might seem like a good idea now, but some time later than `libnl-3.so.200` will become `libnl-3.so.201` and everything will break, we are going to regret it (to say at least).

One other thing, if you're using a package manager you're sure dependencies are satisfied (say if libprotobuf will suddely require libxyz, package manager will take care of it), and if you're doing it manually, it's a time bomb.

So, please revert to installing needed deps during runtime.

Reply:
Done.